### PR TITLE
chore: improve vscode extension debug dx

### PR DESCRIPTION
**Pull request by nhedger**
## Summary

This PR introduces a VS Code configuration that aims to improve the development experience of the Biome extension.

- [x] Adds a debug launch configuration
- [x] Adds a watch task that the debug launch depends on

https://github.com/biomejs/biome/assets/649677/302c8639-291c-49c8-a372-57ca4ef4a5ff

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"type": "npm",
+			"script": "watch",
+			"isBackground": true,
+			"problemMatcher": "$ts-esbuild-watch",
+            "options": {
+                "cwd": "${workspaceFolder}/editors/vscode"
+            },
```
Comment (ematipico) [inline .vscode/tasks.json:11]: it seems there are mixed spaces and tabs

**Review by ematipico — APPROVED** [review]
Do you plan to update the documentation of the extension?

**Comment by nhedger** [comment]
> Do you plan to update the documentation of the extension?

Yes, I do.

File: .vscode/tasks.json (hunk 1 above)
Comment (nhedger) [reply .vscode/tasks.json:11]: Good catch, thanks

**Review by Conaclos — APPROVED** [review]

**Pull request merged**